airflow/jobs/scheduler_job.py

```diff
@@ -1159,10 +1159,7 @@ def _schedule_dag_run(
                 msg='timed_out',
             )
 
-            # Send SLA & DAG Success/Failure Callbacks to be executed
-            self._send_dag_callbacks_to_processor(dag, callback_to_execute)
```

Comment:
Whoops, didn't this end up with the callback being sent twice before too?

Reply:
Yet another case where our session handling causes bugs. Our use of `@create_session` which does an expunge_all _and_ closes the session when it exits needs fixing.

Reply:
No, the callback is executed once, because the `callback` it returned was `None`.